numpy/lib/tests/test_function_base.py

```diff
@@ -2023,6 +2023,12 @@ def test_extreme(self):
         assert_array_almost_equal(c, np.array([[1., -1.], [-1., 1.]]))
         assert_(np.all(np.abs(c) <= 1.0))
 
+    @pytest.mark.parametrize("test_type", [np.half, np.single, np.double, np.longdouble])
+    def test_corrcoef_dtype(self, test_type):
+        cast_A = self.A.astype(test_type)
+        res = corrcoef(cast_A, dtype=test_type)
+        assert_equal(test_type, res.dtype)
```

Comment:
@rossbar: I'd probably be tempted to have left this as `assert test_type == res_dtype` - now that we have pytest, there's little reason to use our test functions for anything other than arrays.

That's not to say it's worth changing it back either, just something to consider for next time :)

Reply:
Thanks for the heads up!